File: Spring_OAuth2/securityserver/src/main/kotlin/com/example/securityserver/model/domain/user/Role.java
package com.example.securityserver.model.domain.user;

import java.io.Serializable;
import java.util.List;

import javax.persistence.*;

@Embeddable
@Entity
@Table(name="ROLE")
public class Role implements Serializable {

    /**
     *
     */
    private static final long serialVersionUID = -2442439253944382064L;

    @Id
    @Column(name = "AUTHORITY_CODE")
    private Long authorityCode;

    @Column(name = "EMP_NO")
    private String empNo;

    @ManyToMany(mappedBy="ROLE")
    private List<User> user;

    public Role() {

    }

    public Long getAuthorityCode() {
        return authorityCode;
    }

    public void setAuthorityCode(Long authorityCode) {
        this.authorityCode = authorityCode;
    }

    public String getEmpNo() {
        return empNo;
    }

    public void setEmpNo(String empNo) {
        this.empNo = empNo;
    }

    public List<User> getUser() {
        return user;
    }

    public void setUser(List<User> user) {
        this.user = user;
    }

    public Role(Long authorityCode, String empNo, List<User> user) {
        this.authorityCode = authorityCode;
        this.empNo = empNo;
        this.user = user;
    }

    @Override
    public String toString() {
        return "Role{" +
                "authorityCode=" + authorityCode +
                ", empNo='" + empNo + '\'' +
                ", user=" + user +
                '}';
    }
}

File: Spring_OAuth2/securityserver/src/main/kotlin/com/example/securityserver/model/domain/user/Role.kt
package com.example.securityserver.model.domain.user;

import java.io.Serializable;
import java.util.List;

import javax.persistence.*;

@Entity
@Table(name = "ROLE")
class Role : Serializable {
    @Id
    @Column(name = "ROLE_AUTHORITY_CODE")
    var roleAuthorityCode: Long? = null

    @Column(name = "EMP_NO")
    var empNo: String? = null

    @ManyToOne
    @JoinColumn(name = "EMP_NO", insertable =false, updatable = false)
    var empBase: EmpBase? = null

    @ManyToOne
    @JoinColumn(name = "ROLE_AUTHORITY_CODE", insertable =false, updatable = false)
    var authority: Authority? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Role

        return roleAuthorityCode == other.roleAuthorityCode
    }

    override fun hashCode(): Int {
        return roleAuthorityCode?.hashCode() ?: 0
    }

    override fun toString(): String {
        return "ROLE(roleAuthorityCode=$roleAuthorityCode, empNo=$empNo, empBase=$empBase, authority=$authority)"
    }
}
